code/modules/atmospherics/machinery/components/unary_devices/condenser.dm

```diff
@@ -0,0 +1,268 @@
+/// How the condensation rate of the condenser scales with part rating
+#define CONDENSER_RATE_BIN_SCALE 0.25
+
+// Leaking:
+/// The ratio of the condenser's current gas it will vent every second.
+#define CONDENSER_VENT_RATIO	0.2
+/// The base rate the condenser will leak reagents at when ruptured
+#define CONDENSER_LEAK_BASE		1	// u/s
+/// The scaling factor for leak rate relative to reagent volume
+#define CONDENSER_LEAK_RATE		0.1	// u/u*s
+/// The scaling factor for leak rate relative to vented gas
+#define CONDENSER_LEAK_SCALE	0.1	// u/n*s
+/// The ratio of vented gas to leaked reagents required to leak foam
+#define LEAK_FOAM_THRESHOLD		2 * REAGENT_MOLE_DENSITY	// n/u
+/// The ratio of vented gas to leaked reagents requires to produce spray
+#define LEAK_SPRAY_THRESHOLD	10 * REAGENT_MOLE_DENSITY	// n/u
+
+/**
+ * # Gas Condenser
+ *
+ * Condenses gases in connected pipes into reagent form.
+ */
+/obj/machinery/atmospherics/components/unary/condenser
+	name = "Gas Condenser"
+	desc = "Condenses gases in connected pipes."
+	icon = 'icons/obj/atmospherics/components/condenser.dmi'
+	base_icon_state = "condenser"
+	layer = OBJ_LAYER
+
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/condenser
+	pipe_flags = PIPING_ONE_PER_TURF
+
+	/// The rate of condensation in this condenser
+	var/condensation_rate_multiplier = 0
+
+
+/obj/machinery/atmospherics/components/unary/condenser/Initialize()
+	. = ..()
+
+	create_reagents(200)
+	AddComponent(/datum/component/plumbing/simple_supply/north)
+
+/obj/machinery/atmospherics/components/unary/condenser/on_construction(obj_color, set_layer)
+	var/obj/item/circuitboard/machine/condenser/board = circuit
+	if(board)
+		piping_layer = board.piping_layer
+		set_layer = piping_layer
+
+	return ..()
+
+/obj/machinery/atmospherics/components/unary/condenser/RefreshParts()
+	. = ..()
+	var/total_bin_rating = 0
+	for(var/obj/item/stock_parts/matter_bin/bin in component_parts)
+		total_bin_rating += bin.get_part_rating()
+
+	condensation_rate_multiplier = total_bin_rating * CONDENSER_RATE_BIN_SCALE
```

Comment:
can't this make infinite gas?

Reply:
That's just a multiplier for the rate the gas condenses at. It doesn't change how much reagent you get out of the gas.